src/Altinn.App.Core/Internal/Validation/ValidationService.cs

```diff
@@ -136,13 +154,16 @@ private Task<List<ValidationIssue>[]> RunDataElementValidators(Instance instance
         ArgumentNullException.ThrowIfNull(dataElement.DataType);
         ArgumentNullException.ThrowIfNull(data);
 
+        using var activity = _telemetry.StartValidateFormDataActivity();
+
         // Locate the relevant data validator services from normal and keyed services
         var dataValidators = _validatorFactory.GetFormDataValidators(dataType.Id)
             .Where(dv => previousData is null || dv.HasRelevantChanges(data, previousData))
             .ToArray();
 
         var issuesLists = await Task.WhenAll(dataValidators.Select(async (v) =>
         {
+            using var activity = _telemetry.StartRunFormDataValidatorActivity();
```

Comment:
I might be confused, but does this functionally replace the `LogDebug` statement on the next line?

Instance id and data element might be tracked at a higher level, but `v.ValidationSource` likely makes a lot of sense to track here (and also instead of `v.GetType().Name` in the `LogDebug` statement.).

Reply:
Logs are separate from traces (`Activity`s are traces), logs are a simple list of messages while traces form a sort of call-graph. Distributed traces are particularly useful for us since we're in a microservices setting.

Fully agree, lifting tags as high up as possible in the trace hierarchy, and including new information as we go down the stack 👍